packages/flutter/lib/src/cupertino/expansion_tile.dart

```diff
@@ -0,0 +1,234 @@
+    super.initState();
+    _tileController = widget.controller ?? ExpansibleController();
+  }
+
+  @override
+  void dispose() {
+    if (widget.controller == null) {
+      _tileController.dispose();
+    }
+    super.dispose();
+  }
+
+  Widget? _buildIcon(BuildContext context, Animation<double> animation) {
+    _iconTurns = animation.drive(_quarterTween.chain(CurveTween(curve: _kAnimationCurve)));
+    final double? size = CupertinoTheme.of(context).textTheme.textStyle.fontSize;
+    // Replicate the Icon logic here to get a slightly bolder icon.
+    return RotationTransition(
+      turns: _iconTurns,
+      child: SizedBox(
+        width: size,
+        height: size,
+        child: Center(
+          child: Text.rich(
+            TextSpan(
+              text: String.fromCharCode(CupertinoIcons.right_chevron.codePoint),
+              style: TextStyle(
+                inherit: false,
+                color: CupertinoColors.activeBlue,
+                fontSize: _kIconFontSize,
+                fontWeight: FontWeight.w900,
+                fontFamily: CupertinoIcons.right_chevron.fontFamily,
+                package: CupertinoIcons.right_chevron.fontPackage,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
+  void _onHeaderTap() {
+    if (_tileController.isExpanded) {
+      _tileController.collapse();
+    } else {
+      _tileController.expand();
+    }
+    _fadeController.show();
+  }
+
+  Widget _buildHeader(BuildContext context, Animation<double> animation) {
+    return CupertinoListTile(
+      key: _headerKey,
+      onTap: _onHeaderTap,
+      title: widget.title,
+      trailing: _buildIcon(context, animation),
+      backgroundColorActivated: CupertinoColors.transparent,
+    );
+  }
+
+  Widget _buildExpansible(
+    BuildContext context,
+    Widget header,
+    Widget body,
+    Animation<double> animation,
+  ) {
+    final Widget child = Column(
+      mainAxisSize: MainAxisSize.min,
+      children: <Widget>[
+        header,
+        Opacity(
+          opacity:
+              animation.isAnimating && widget.transitionMode == ExpansionTileTransitionMode.fade
+                  ? 0.0
+                  : 1.0,
+          child: body,
+        ),
+      ],
+    );
+
+    if (widget.transitionMode == ExpansionTileTransitionMode.scroll) {
+      return child;
+    }
+
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
+        return OverlayPortal(
+          controller: _fadeController,
+          overlayChildBuilder: (BuildContext context) {
+            final BuildContext headerContext = _headerKey.currentContext!;
+            final RenderBox overlay =
+                Overlay.of(headerContext).context.findRenderObject()! as RenderBox;
+            final RenderBox headerBox = headerContext.findRenderObject()! as RenderBox;
+            final Offset headerOffset = headerBox.localToGlobal(Offset.zero, ancestor: overlay);
```

Comment:
maybe this is not too big of a deal, if the overlay is open and scrolling at the same time. will there be visual glitches?

Reply:
oh nvm, didnt see the previous comment, I think this is fine then